src/routes/connect/Connect.svelte

```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+    import { isPrivateKeyVisible, setInputValue, setConnectionError } from './connectStore';
+    import { getStoredKey, getStellarKeypair, storeKey } from './connectHelpers';
+    import { publicKey } from '../../store/store';
+    import InvalidPrivateKeyError from './errors/InvalidPrivateKeyError';
+    import StorageKeyNotFoundError from './errors/StorageKeyNotFoundError';
+
+    (async function connectWithStorage(): Promise<void> {
+        try {
+            const privateKey = await getStoredKey();
+            connectWithSecretKey(privateKey);
+        } catch (e) {
+            if (e instanceof StorageKeyNotFoundError) {
+                console.log('No key was found in storage');
+            }
+        }
+    })();
+
+    async function connectWithSecretKey(key: string): Promise<void> {
+        try {
+            const stellarKeyPair = await getStellarKeypair(key);
+            storeKey(key);
+            $publicKey = stellarKeyPair.publicKey();
+        } catch (e) {
+            if (e instanceof InvalidPrivateKeyError) {
+                $publicKey = '';
+                $setConnectionError = 'Invalid key, please try again';
+            }
+        }
+    }
```

Comment:
functions should be declared before they are used, you use this in the function above, so you should put this function first